test/e2e/basic_test.go

```diff
@@ -0,0 +1,207 @@
+		_, err = clientset.Services(ns.Name).Create(serverSvc)
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		ginkgo.By("deploying netchecker agent daemonset")
+		var ncAgentLabels = map[string]string{"app": "netchecker-agent"}
+		serverEndpointArg := fmt.Sprintf("--serverendpoint=netchecker-service:%d", serverPort)
+		cmd := []string{"netchecker-agent", "--alsologtostderr=true", "--v=10", serverEndpointArg, "--reportinterval=20"}
+		agentDS := newDaemonSet("netchecker-agent", "netchecker-agent", "mirantis/k8s-netchecker-agent",
+			[]string{"sh", "-c", strings.Join(cmd, " ")}, ncAgentLabels, false, true,
+			[]v1.EnvVar{
+				{Name: "MY_NODE_NAME", ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "spec.nodeName"}}},
+				{Name: "MY_POD_NAME", ValueFrom: &v1.EnvVarSource{FieldRef: &v1.ObjectFieldSelector{FieldPath: "metadata.name"}}},
+			},
+		)
+		_, err = clientset.Extensions().DaemonSets(ns.Name).Create(agentDS)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		testutils.Logf("current pods: %v\n", getPods(clientset, ns))
+		testutils.Logf("current services: %v\n", getServices(clientset, ns))
+
+		ginkgo.By("veryfiying that service is reachable using DNS")
+		//verifyServiceReachable(serverPort, []string{"0.0.0.0"}...)
+	})
+})
+
+func newPrivilegedPodSpec(containerName, imageName string, cmd []string, hostNetwork, privileged bool, env []v1.EnvVar) v1.PodSpec {
+	return v1.PodSpec{
+		HostNetwork: hostNetwork,
+		Containers: []v1.Container{
+			{
+				Name:            containerName,
+				Image:           imageName,
+				Command:         cmd,
+				SecurityContext: &v1.SecurityContext{Privileged: &privileged},
+				ImagePullPolicy: v1.PullIfNotPresent,
+				Env:             env,
+			},
+		},
+	}
+}
+
+func newPod(podName, containerName, imageName string, cmd []string, labels map[string]string, hostNetwork bool, privileged bool, env []v1.EnvVar) *v1.Pod {
+	return &v1.Pod{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:   podName,
+			Labels: labels,
+		},
+		Spec: newPrivilegedPodSpec(containerName, imageName, cmd, hostNetwork, privileged, env),
+	}
+}
+
+func newDaemonSet(dsName, containerName, imageName string, cmd []string, labels map[string]string, hostNetwork, privileged bool, env []v1.EnvVar) *v1beta1.DaemonSet {
+	return &v1beta1.DaemonSet{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:   dsName,
+			Labels: labels,
+		},
+		Spec: v1beta1.DaemonSetSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: meta_v1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: newPrivilegedPodSpec(containerName, imageName, cmd, hostNetwork, privileged, env),
+			},
+		},
+	}
+
+}
+
+func newDeployment(deploymentName string, replicas int32, podLabels map[string]string, imageName string, image string, cmd []string, env []v1.EnvVar) *v1beta1.Deployment {
+	return &v1beta1.Deployment{
+		ObjectMeta: meta_v1.ObjectMeta{Name: deploymentName},
+		Spec: v1beta1.DeploymentSpec{
+			Replicas: &replicas,
```

Comment:
It is just to be in accordance with DeploymentSpec specification: "Replicas *int32".